tests/unit/tf/blocks/test_interactions.py

```diff
@@ -26,7 +28,66 @@ def test_fm_pairwise_interaction():
 
     inputs = tf.random.uniform((NUM_ROWS, NUM_FEATS, EMBED_DIM))
 
-    pairwise_interaction = ml.FMPairwiseInteraction()
+    pairwise_interaction = mm.FMPairwiseInteraction()
     outputs = pairwise_interaction(inputs)
 
     assert list(outputs.shape) == [NUM_ROWS, EMBED_DIM]
+
+
+def test_fm_block(ecommerce_data: Dataset):
+    schema = ecommerce_data.schema
+
+    fm_block = mm.FMBlock(
+        schema,
+        factors_dim=32,
+    )
+
+    batch = mm.sample_batch(ecommerce_data, batch_size=16, include_targets=False)
+    output = fm_block(batch)
+    output.shape.as_list() == [16, 1]
+
+
+def test_fm_block_with_multi_hot_categ_features(testing_data: Dataset):
+    schema = testing_data.schema
+    cat_schema = schema.select_by_tag(Tags.CATEGORICAL)
+    cat_schema_onehot = cat_schema.remove_by_tag(Tags.LIST)
+    cat_schema_multihot = cat_schema.select_by_tag(Tags.LIST)
+
+    input_block = mm.InputBlockV2(
+        schema,
+        embeddings=mm.Embeddings(
+            cat_schema,
+            dim=32,
+            sequence_combiner="mean",
+        ),
+        aggregation=None,
+    )
+
+    wide_input_block = mm.ParallelBlock(
+        {
+            "categorical_ohe": mm.SequentialBlock(
+                mm.Filter(cat_schema_onehot),
+                mm.CategoryEncoding(cat_schema_onehot, sparse=True, output_mode="one_hot"),
```

Comment:
If `mm.CategoryEncoding` already takes in a schema, can’t we make that class also do the filtering?

Reply:
The thing is that `CategoryEncoding` can be used with both `output_mode="one_hot"` and `output_mode="multi_hot"`. 
I am not sure we should trust within `CategoryEncoding` that multi-hot features will be tagged as SEQUENCE / LIST. So I used the test as an example on how the use can ensure the right features are selected for each branch